tests/test_main.py

```diff
@@ -570,7 +572,8 @@ def test_with(self):
                 a(b(c()))
 
     def test_listcomp(self):
-        if sys.version_info >= (3, 11):
+        if (3, 11) <= sys.version_info < (3, 12):
+            # ListComp is inlined in 3.12
             result = [calling_expression() for e in [1]]
```

Comment:
It's not important, but it's surprising to me that inlining comprehensions made identifying the node not work.

Reply:
good point. I found some missing tests and added some rules for __iter__.
__iter__ is the only function I can think of where the user could be able to use executing to get an ast-node.

Reply:
Oh, I understand what's happening now.